doc/source/whatsnew/v1.0.0.rst

```diff
@@ -38,7 +38,28 @@ Backwards incompatible API changes
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
 - :class:`pandas.core.groupby.GroupBy.transform` now raises on invalid operation names (:issue:`27489`).
--
+- :class:`pandas.core.arrays.IntervalArray` adopts a new ``__repr__`` in accordance with other array classes (:issue:`25022`)
+
+*pandas 0.25.x*
+
+.. code-block:: ipython
+
+   In [1]: pd.arrays.IntervalArray.from_tuples([(0, 1), (2, 3)])
+   Out[1]:
+   IntervalArray([(0, 1], (2, 3]],
+                 closed='right',
+                 dtype='interval[int64]')
+
+*pandas 1.0.0*
+
+.. code-block:: ipython
```

Comment:
Sorry could have been clearer but just make this `.. ipython:: python`` for the directive. You shouldn't then need the In / Out below and it will render automatically for you (this is typically what we do with new behavior)